msal/application.py

```diff
@@ -140,7 +158,8 @@ def _build_client(self, client_credential, authority):
                 client_credential["private_key"], algorithm="RS256",
                 sha1_thumbprint=client_credential.get("thumbprint"), headers=headers)
             client_assertion = signer.sign_assertion(
-                audience=authority.token_endpoint, issuer=self.client_id)
+                audience=authority.token_endpoint, issuer=self.client_id,
+                additional_claims=self.client_claims or {})
```

Comment:
I think we can just pass the claims here because we are anyway doing "or empty dictionary" assignment inside the assertion function again. Any thoughts ?

Reply:
We discussed offline. Even though the underlying internal api HAPPENS TO also accepts a `None` as input, this higher level code snippet does not necessarily want to depend on that behavior when we don't have to.